lib/core/src/rodsPath.cpp

```diff
@@ -265,6 +265,27 @@ addSrcInPath( rodsPathInp_t *rodsPathInp, const char *inPath ) {
         return USER__NULL_INPUT_ERR;
     }
 
+    // Issue #3658 - https://github.com/irods/irods/issues/3658
```

Comment:
entire comment should just be:
`This section removes trailing slashes from the incoming directory path in an invoking icommand (e.g. ireg).`

Reply:
Theses are my opinions:  

The function addSrcInPath() is called from parseCmdLinePath() which is used by a total of nineteen icommands.   Since this removal of a trailing slash from the directory path affects so many utilities, I think it's a good idea to be very verbose here about what's being changed and why. 

I don't think that assuming that the developer looking at this would figure out to go look at the source code control history, or at the git issue detail, is a reasonable assumption. 

In addition, while doing this we (Jason, you, me) discussed the issue of normalizing the treatment of all paths in the future.  We agreed (or I thought we did) that this (removal of trailing slashes in various places in the code) deserves a TODO section that would remind the developers who are doing the normalizing, to clean up (what would then be) unnecessary editing of the path string in various places in the code.  This comment also relates to your comments in rsPhyPathReg.cpp above.

Give me the final verdict, and I'll follow your direction.  Thanks.